api/base/views.py

```diff
@@ -6,57 +6,214 @@
+    When a request fails for whatever reason, the OSF API will return an appropriate HTTP error code and include a
+    descriptive error in the body of the response.  The response body will be an object with a key, `errors`, pointing
+    to an array of error objects.  Generally, these error object will consist of a `detail` key with a detailed error
+    message, but may include additional information in accordance with the [JSON-API error
+    spec](http://jsonapi.org/format/1.0/#error-objects).
 
-        ##General API Usage
-        Each endpoint will have its own documentation, but there are some general principles.
+    ##OSF Enum Fields
 
-        ###Filtering
-        Collections can be filtered by adding a query parameter in the form:
+    Some entities in the OSF API have fields that only take a restricted set of values.  Those fields are listed here
+    for reference.  Fuller descriptions are available on the relevent entity pages.
 
-            filter[<fieldname>]=<matching information>
-        For example, if you were trying to find [Lise Meitner](http://en.wikipedia.org/wiki/Lise_Meitner):
+    ###OSF Node Categories
 
-            /users?filter[fullname]=meitn
-        You can filter on multiple fields, or the same field in different ways, by &-ing the query parameters together.
+        value                 description
+        ------------------------------------------
+        project               Project
+        hypothesis            Hypothesis
+        methods and measures  Methods and Measures
+        procedure             Procedure
+        instrumentation       Instrumentation
+        data                  Data
+        analysis              Analysis
+        communication         Communication
+        other                 Other
 
-            /users?filter[fullname]=lise&filter[family_name]=mei
-        ###Links
-        Responses will generally have associated links which are helpers to keep you from having to construct URLs in
-        your code or by hand. If you know the route to a high-level resource, you can go to that route. For example:
+    ###OSF Node Permission keys
 
-            /nodes/<node_id>
+        value        description
+        ------------------------------------------
+        read         Read-only access
+        write        Write access (make changes, cannot delete)
+        admin        Admin access (full write, create, delete, contributor add)
 
-        is a good route to create rather than going to /nodes/ and navigating by id filtering. However, if you are
-        creating something that crawls the structure of a node going to the child node or gathering children,
-        contributors, and similar related resources, then take the link from the object you\'re crawling rather than
-        constructing the link yourself.
+    ###Storage Providers
 
-        In general, links include:
+    Valid storage providers are:
```

Comment:
I don't think there's anything to do about this right now, but we might want to think about a way to automate the creation of this list. That or just ensure that any PRs that add a storage provider update this list.